RFS/src/main/java/com/rfs/common/RestClient.java

```diff
@@ -28,108 +25,61 @@ public Response(int responseCode, String responseBody, String responseMessage) {
-                responseBody = br.lines().collect(Collectors.joining(System.lineSeparator()));
-            }
-        } else {
-            // Read input stream for successful requests
-            try (BufferedReader br = new BufferedReader(new InputStreamReader(conn.getInputStream(), StandardCharsets.UTF_8))) {
-                responseBody = br.lines().collect(Collectors.joining(System.lineSeparator()));
-            }
-        }
-
-        if (quietLogging || (responseCode == HttpURLConnection.HTTP_CREATED || responseCode == HttpURLConnection.HTTP_OK)) {
-            logger.debug("Response Code: " + responseCode + ", Response Message: " + conn.getResponseMessage() + ", Response Body: " + responseBody);
-        } else {
-            logger.error("Response Code: " + responseCode + ", Response Message: " + conn.getResponseMessage() + ", Response Body: " + responseBody);
-        }
-
-        conn.disconnect();
 
-        return new Response(responseCode, responseBody, conn.getResponseMessage());
+        this.client = HttpClient.create()
+            .baseUrl(connectionDetails.url)
+            .headers(h -> {
+                h.add("Content-Type", "application/json");
+                if (connectionDetails.authType == ConnectionDetails.AuthType.BASIC) {
+                    String credentials = connectionDetails.username + ":" + connectionDetails.password;
+                    String encodedCredentials = Base64.getEncoder().encodeToString(credentials.getBytes());
+                    h.add("Authorization", "Basic " + encodedCredentials);
+                }
+            });
     }
-    
-    public Response put(String path, String body, boolean quietLogging) throws Exception {
-        String urlString = connectionDetails.url + "/" + path;
-        
-        URL url = new URL(urlString);
-        HttpURLConnection conn = (HttpURLConnection) url.openConnection();
 
-        // Set the request method
-        conn.setRequestMethod("PUT");
-
-        // Set the necessary headers
-        conn.setRequestProperty("Content-Type", "application/json");
-        setAuthHeader(conn);
-
-        // Enable input and output streams
-        conn.setDoOutput(true);
-
-        // Write the request body
-        try(OutputStream os = conn.getOutputStream()) {
-            byte[] input = body.getBytes("utf-8");
-            os.write(input, 0, input.length);           
-        }        
+    public Mono<Response> getAsync(String path, boolean quietLogging) {
+        return client.get()
+            .uri("/" + path)
+            .responseSingle((response, bytes) -> bytes.asString()
+                .map(body -> new Response(response.status().code(), body, response.status().reasonPhrase()))
+                .doOnNext(resp -> logResponse(resp, quietLogging)));
+    }
 
-        // Report the results
-        int responseCode = conn.getResponseCode();
+    public Response get(String path, boolean quietLogging) {
```

Comment:
where do you use the blocking calls?  Given that it looks like everything is async, why not go all the way :)

Reply:
We use the blocking calls where performance isn't important but ordering is - like whether to do a PUT based on the response from a GET.

Reply:
Shouldn't you be chaining those calls?  I'm not sure how the threading model works with reactor, but with a netty client, all IO for a given channel runs on one thread.  Blocking on that thread can have profound, negative consequences (deadlock being one of them)